scripts/utils/bin/game_monitor.rs

```diff
@@ -0,0 +1,336 @@
+use alloy_primitives::{Address, U256};
+use alloy_provider::ProviderBuilder;
+use anyhow::{Context, Result};
+use clap::Parser;
+use fault_proof::contract::{
+    DisputeGameFactory::DisputeGameFactoryInstance, OPSuccinctFaultDisputeGame,
+};
+use log::{error, info, warn};
+use std::{
+    collections::{HashMap, HashSet},
+    env, fs,
+    fs::File,
+    path::{Path, PathBuf},
+    process::{Child, Command, Stdio},
+    time::Duration,
+};
+use tokio::time::sleep;
+
+const GAME_TYPE: u32 = 42;
+
+/// Arguments for the game monitor.
+#[derive(Debug, Clone, Parser)]
+pub struct GameMonitorArgs {
+    /// The environment file to use. This file should contain the following environment variables:
+    ///
+    /// - DISPUTE_GAME_FACTORY_ADDRESS: The address of the dispute game factory contract.
+    ///
+    /// - L1_RPC: The URL of the L1 RPC endpoint.
+    ///
+    /// - L1_BEACON_RPC: The URL of the L1 beacon RPC endpoint.
+    ///
+    /// - L2_RPC: The URL of the L2 RPC endpoint.
+    ///
+    /// - L2_NODE_RPC: The URL of the L2 node RPC endpoint.
+    ///
+    /// - EIGENDA_PROXY_ADDRESS: The address of the eigenda proxy service.
+    ///
+    /// - OP_SUCCINCT_MOCK: Must be 'true'
+    ///
+    /// - SP1_PROVER: Must be 'mock'
+    #[arg(long, default_value = ".env")]
+    pub env_file: PathBuf,
+
+    /// The polling interval in seconds.
+    #[arg(long, default_value = "30")]
+    pub poll_interval: u64,
+
+    /// Maximum number of concurrent cost estimator processes.
+    #[arg(long, default_value = "5")]
+    pub max_concurrent: usize,
+
+    /// The path to the cost estimator binary.
+    #[arg(long, default_value = "cost-estimator")]
+    pub cost_estimator_binary_path: PathBuf,
+
+    /// The directory under which to store the logs.
+    #[arg(long, default_value = "logs")]
+    pub logs_dir: PathBuf,
+}
+
+/// Represents a running cost estimator process for a game.
+struct RunningEstimator {
+    process: Child,
+    log_file: PathBuf,
+}
+
+/// Tracks the state of the game monitor.
+struct MonitorState {
+    /// Set of game addresses we've already spawned estimators for.
+    processed_games: HashSet<Address>,
+    /// Currently running estimator processes.
+    running_processes: HashMap<u64, RunningEstimator>,
+    /// The last game index we checked.
+    last_checked_index: Option<u64>,
+}
+
+impl MonitorState {
+    fn new() -> Self {
```

Comment:
As far as I can tell you could just derive `Default` and use that instead of implementing this yourself.